### DISCUSSION TITLE
Rule of disonnection

### DISCUSSION THREAD

**Discussion by Ruslan-I7** [NodeRedis/node_redis#2451] [Q&A] [answered]
Guys hello,

could u tell, how to properly disconnect redis 6 in express? Does it make sense to close the connection in each request?

What exactly method to use? `CLIENT KILL`, `disconnect` or something else? `Quit` is deprecated, so I'm not looking into this option.

And one more, if I use Prisma library where is said that 

> automatically connects when you run your first query, creates a [connection pool and disconnects when the Node.js process ends. 

Maybe it doesn't make sense to make a disconnection of redis at all?

**Comment by leibale** [comment, marked as answer]
In most cases, you'll want to create one global client and never close it (unless the process closes, which is done automatically)

**Reply by ivan-kleshnin** [reply]
`connect` prevents the Node/Bun process from exiting. So what should we do to disconnect?
 I hope you're not proposing to call `process.exit`? 😨 

P.S
I'm using Redis in a CLI script. It's not a server process and it should terminate normally.

**Comment by Ruslan-I7** [comment]
Thanks, I thought so too. But one guy said that he made disconnections after each request and I decided to clarify it. He had a lot of requests and requests were connected and disconnected redis very often. It made me confused
